app/_src/gateway/key-concepts/routes/index.md

```diff
@@ -15,6 +15,96 @@ For example, if you have an external application and an internal application tha
 
 In the example above, two routes can be created, say `/external` and `/internal`, and both routes can point to `example_service`. A policy can be configured to limit how often the `/external` route is used and the route can be communicated to the external client for use. When the external client tries to access the service via {{site.base_gateway}} using `/external`, they are rate limited. But when the internal client accesses the service using {{site.base_gateway}} using `/internal`, the internal client will not be limited.
 
+## How {{site.base_gateway}} routes requests in `traditional_compat` mode
+
+For each incoming request, {{site.base_gateway}} needs to determine
+which service gets to handle it based on the routes that have been
+defined.  We describe how the routing process works in
+`traditional_compat` mode, in a logical sense, below.  The actual
+router implementation optimizes the process and caches results to
+avoid having to scan all routes repeatedly.  Note that when using the
+`expressions` router, route priorities are explicitly defined and this
+description does not apply.  Underneath, routes are converted from the
+`traditional_compat` format to the `expressions` format.
+
+In general, the router uses the highest priority matching route to
+handle a request.  If there are multiple matching routes with the same
+priority, it is not defined which of the two routes will be used.
+
+In `traditional_compat` mode, the priority of a route is determined as
+follows, by the order of descending significance:
```

Comment:
Perhaps use an ordered list instead of an unordered one here?

```
1. Priority points
2. Wildcard hosts
3. Header count
4. Regular expressions and prefix paths
```